### ISSUE TITLE
The variable binding is not working on debug (for object enum variables)

### ISSUE THREAD

**Issue by limonadev** [peiffer-innovations/json_dynamic_widget#10]
In the README [this line](https://github.com/peiffer-innovations/json_dynamic_widget/blame/main/README.md#L148) specifies that every `arg` can be a `{{variable}}`. But in debug mode, when trying to use a variable set in the Registry the program throws an exception.
It only fails in debug because the schema validation only works in debug mode.

## Steps to reproduce (A simple example below):
1. Add a `set_value` widget into one json, setting one `{{customVariable}}`
2. Replace any `object` with enum arg (except Color) with the variable syntax `{{customVariable}}` (tested with brightness and alignment)
3. Run the example and go into the page of the modified json

```json
{
  "type": "set_value",
  "args": {
    "customBrightness": "dark"
  },
  "child": {
    "type": "scaffold",
    "args": {
      "appBar": {
        "type": "app_bar",
        "args": {
          "title": {
            "type": "text",
            "args": {
              "text": "Test"
            }
          },
          "brightness": "{{customBrightness}}"
        }
      }
    }
  }
}
```

**Comment by jpeiffer** [comment]
Hmmm... this is a hard one.  The problem is that either the schema validator needs to accept any arbitrary string, which defeats the values the enums in the schema provide, or the schema validator will always fail in this case.

I need to think about which provides more overall value.  The more specific schema and better support for JSON Schema toolsets, or fixing this failed validation.

In the mean time, developers who need to can bypass this issue by calling:
```json
void main() {
  SchemaValidator().enabled = false;
  ...
}
```

**Comment by limonadev** [comment]
Also related, when using a variable `{{custom}}` to encode some object (not bool, number nor string) in some widget, if the builder uses the function `ThemeDecoder.decodeSomething()`, the program throws an exception because the variable is already the object, but the `decodeSomething()` function usually treats internally the `value` argument as a `Map`.

## Steps to reproduce:
1. Register some custom variable in the Registry (I registered a `Rect` variable).
2. Use that variable in any json (I used it in `asset_images.json`)

**Comment by limonadev** [comment]
As a proposal to solve [this](https://github.com/peiffer-innovations/json_dynamic_widget/issues/10#issuecomment-680116851) part of the issue, I've tested this new class to replace each call to `ThemeDecoder.decodeSomething()` where it wants to be used as a {{variable}} or a JSON map structure. This doesn't solve the object enum variables.

This is the basic class, only with one function (following the "Steps to reproduce", I solved the problem for `Rect` variables but it also can be generalized to all the objects)
```dart
import 'package:flutter/painting.dart';
import 'package:json_theme/json_theme.dart';

class InstanceDecoder {
  static Rect decodeRect(
    dynamic instance, {
    Rect defaultInstance,
    bool validate = true,
  }) {
    Rect result;

    if (instance is Rect) {
      result = instance;
    } else {
      result = ThemeDecoder.decodeRect(
        instance,
        validate: validate,
      );
    }

    result ??= defaultInstance;

    return result;
  }
}
```

**Comment by jpeiffer** [comment]
Please separate this out into a different defect.  It's not really the same problem as the original one.

**Comment by limonadev** [comment]
> In the mean time, developers who need to can bypass this issue by calling:
> 
> ```dart
> void main() {
>   SchemaValidator().enabled = false;
>   ...
> }
> ```

This bypass doesn't work as expected. The exception is caused inside `json_dynamic_widget` package, not the `json_theme`. So, to disable validation it's necessary to use a custom `JsonWidgetRegistry` instead of `JsonWidgetRegistry.instance`:
```dart
var registry = JsonWidgetRegistry(
    debugLabel: 'Disabled Validation',
    disableValidation: true,
  );
```

**Comment by jpeiffer** [comment]
I did some more research on the capabilities of JSON Schemas and found a good solution that allowed keeping the enum values while also allowing for the dynamic replacements.

This is fully fixed in `1.0.4`.  The actual fix was in `json_theme`, and the "fix" here was just updating the dependency to pull in the most recent `json_theme` version + adding this example to the example app itself.
